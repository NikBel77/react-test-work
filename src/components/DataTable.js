import React from 'react'
import TableRow from './TableRow'

import caretRight from '../icons/caret-r.svg'

class DataTable extends React.Component {
    constructor(props) {
        super(props);
        this.handleIdChange = this.handleIdChange.bind(this)
    }

    handleIdChange(id) {
        this.props.onSelectPerson(id)
    }

    render() {
        return (
            <div className='container'>
                <table className="table table-hover table-sm table-dark">
                    <thead>
                        <tr>
                            <th scope="col">
                                id<img className='table-caret' src={caretRight} alt='caret'/>
                            </th>
                            <th scope="col">
                                First name<img className='table-caret' src={caretRight} alt='caret'/>
                            </th>
                            <th scope="col">
                                Last name<img className='table-caret' src={caretRight} alt='caret'/>
                            </th>
                            <th scope="col">
                                Email<img className='table-caret' src={caretRight} alt='caret'/>
                            </th>
                            <th scope="col">
                                Phone<img className='table-caret' src={caretRight} alt='caret'/>
                            </th>
                        </tr>
                    </thead>
                    <tbody>
                        {this.props.dataList.map((personData, i) => {
                            const isSelected = this.props.activeId === personData.id;

                            return (
                                <TableRow
                                    onIdChange={this.handleIdChange}
                                    personData={personData}
                                    isSelected={isSelected}
                                    key={i} />
                            )
                        })}
                    </tbody>
                </table>
            </div>
        )
    }
}

export default DataTable