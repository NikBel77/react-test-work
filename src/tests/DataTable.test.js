import React from 'react'
import ReactDOM from 'react-dom'
import DataTable from '../components/DataTable'
import renderer from 'react-test-renderer'


const testData = [
    {
        id: 101,
        firstName: 'Sue',
        lastName: 'Corson',
        email: 'DWhalley@in.gov',
        phone: '(612)211-6296',
        address: {
            streetAddress: '9792 Mattis Ct',
            city: 'Waukesha',
            state: 'WI',
            zip: '22178'
        },
        description: 'et lacus magna dolor...',
    }
]

test('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <DataTable dataList={testData}
            activePerson={testData[0]}
            activeColumn={'id'}
        />, div);
    ReactDOM.unmountComponentAtNode(div);
});
test('snapshot', () => {
    const tree = renderer.create(<DataTable
        dataList={testData}
        activePerson={testData[0]}
        activeColumn={'id'}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});