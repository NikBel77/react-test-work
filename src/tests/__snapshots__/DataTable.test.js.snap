// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <table
    className="table table-hover table-sm table-dark"
  >
    <thead>
      <tr>
        <th
          onClick={[Function]}
          scope="col"
        >
          ID
          <img
            alt="caret"
            className="table-caret"
            src="caret-r.svg"
          />
        </th>
        <th
          onClick={[Function]}
          scope="col"
        >
          First name
          <img
            alt="caret"
            className="table-caret"
            src="caret-r.svg"
          />
        </th>
        <th
          onClick={[Function]}
          scope="col"
        >
          Last name
          <img
            alt="caret"
            className="table-caret"
            src="caret-r.svg"
          />
        </th>
        <th
          onClick={[Function]}
          scope="col"
        >
          Email
          <img
            alt="caret"
            className="table-caret"
            src="caret-r.svg"
          />
        </th>
        <th
          onClick={[Function]}
          scope="col"
        >
          Phone
          <img
            alt="caret"
            className="table-caret"
            src="caret-r.svg"
          />
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="bg-danger"
        onClick={[Function]}
      >
        <td>
          101
        </td>
        <td>
          Sue
        </td>
        <td>
          Corson
        </td>
        <td>
          DWhalley@in.gov
        </td>
        <td>
          (612)211-6296
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
